include/nanobind/ndarray.h

```diff
@@ -293,9 +293,12 @@ template <typename... Args> class ndarray {
     }
 
     dlpack::dtype dtype() const { return m_dltensor.dtype; }
+    size_t size() const { return Info::size; }
     size_t ndim() const { return m_dltensor.ndim; }
     size_t shape(size_t i) const { return m_dltensor.shape[i]; }
     int64_t stride(size_t i) const { return m_dltensor.strides[i]; }
+    auto* shape_ptr() const { return &m_dltensor.shape[0]; }
```

Comment:
Let's not use `auto` in a public interface especially when the type is easily knowable. You can return the array directly and don't need `&array[0]`.

Reply:
Done. I'm just a bit confused about the type of `shape_ptr`. From the other return types I would have expected it to be `size_t*`. However, my compiler is convinced about it being `int64_t*`